src/SyncMe/SyncMe.Common/ISyncEventsRepository.cs

```diff
@@ -1,11 +1,19 @@
-﻿using SyncMe.Models;
+﻿using System.Linq.Expressions;
+using SyncMe.Models;
+using SyncMe.Queries;
 
 namespace SyncMe;
 
 public interface ISyncEventsRepository
 {
-    IReadOnlyCollection<SyncEvent> GetAllSyncEvents();
     bool TryGetSyncEvent(Guid id, out SyncEvent syncEvent);
-    Guid AddSyncEvent(SyncEvent syncEvent);
-    void RemoveEvents(Func<SyncEvent, bool> predicate);
+    SyncEvent AddSyncEvent(SyncEvent syncEvent);
+    void RemoveEvents(Expression<Func<SyncEvent, bool>> predicate);
+    void RemoveEvent(Guid eventId);
+    IReadOnlyCollection<SyncEvent> GetByNamespace(string namespaceKey);
+    void UpdateEvents(IEnumerable<SyncEvent> syncEvents);
+    IReadOnlyCollection<SyncEvent> GetNotRepeatableEvents(SyncEventQuery query);
+    IReadOnlyCollection<SyncEvent> GetEveryYearRepeatableEvents(SyncEventQuery query);
+    IReadOnlyCollection<SyncEvent> GetRepeatableLessMonthEvents(SyncEventQuery query);
```

Comment:
What's the intent of this method? A bit strange name

Reply:
I know, I'd like to refactor it to divide all types of Repeatability of events. The meaning of this name is repeatability frequency less than month